modules/opencode/hm.nix

```diff
@@ -0,0 +1,254 @@
+{ mkTarget, ... }:
+mkTarget {
+  name = "opencode";
+  humanName = "opencode";
+
+  configElements =
+    { colors }:
+    {
+      programs.opencode =
+        let
+          theme = "stylix";
+        in
+        {
+          settings = { inherit theme; };
+          themes = {
+            ${theme} = {
+              "$schema" = "https://opencode.ai/theme.json";
+              defs = {
+                inherit (colors.withHashtag)
+                  base00
+                  base01
+                  base02
+                  base03
+                  base04
+                  base05
+                  base06
+                  base07
+                  base08
+                  base09
+                  base0A
+                  base0B
+                  base0C
+                  base0D
+                  base0E
+                  base0F
+                  ;
+              };
```

Comment:
> # Color definitions
>
> The `defs` section is optional and it allows you to define reusable colors that can be referenced in the theme.
>
> -- https://opencode.ai/docs/themes/#color-definitions

Since we use Nix, we do not rely on its DSL for variable declarations. To align with the other Stylix modules would it not be better to directly use the variable? For example:

```diff
                 accent = {
-                  dark = "base07";
-                  light = "base07";
+                  dark = colors.withHashtag.base07;
+                  light = colors.withHashtag.base07;
                 };
```

Reply:
Although we plan on removing such `with` uses, I am fine with either approach until we do a treewide migration.

Reply:
Well, it is a small change with multicursors, so I changed it to exclude the `with` statement. I usually don't like it, but in such cases, it is convenient since there are no additional variables with whom they can be confused.